internal/services/containers/container_registry_task_resource.go

```diff
@@ -0,0 +1,1573 @@
+}
+
+type ContainerRegistryTaskModel struct {
+	Name                string                 `tfschema:"name"`
+	ContainerRegistryId string                 `tfschema:"container_registry_id"`
+	Identity            []Identity             `tfschema:"identity"`
+	AgentConfig         []AgentConfig          `tfschema:"agent_setting"`
+	AgentPoolName       string                 `tfschema:"agent_pool_name"`
+	IsSystemTask        bool                   `tfschema:"is_system_task"`
+	LogTemplate         string                 `tfschema:"log_template"`
+	Platform            []Platform             `tfschema:"platform_setting"`
+	Enabled             bool                   `tfschema:"enabled"`
+	TimeoutInSec        int                    `tfschema:"timeout_in_seconds"`
+	DockerStep          []DockerStep           `tfschema:"docker_step"`
+	FileTaskStep        []FileTaskStep         `tfschema:"file_task_step"`
+	EncodedTaskStep     []EncodedTaskStep      `tfschema:"encoded_task_step"`
+	BaseImageTrigger    []BaseImageTrigger     `tfschema:"base_image_trigger"`
+	SourceTrigger       []SourceTrigger        `tfschema:"source_trigger"`
+	TimerTrigger        []TimerTrigger         `tfschema:"timer_trigger"`
+	RegistryCredential  []RegistryCredential   `tfschema:"registry_credential"`
+	Tags                map[string]interface{} `tfschema:"tags"`
+}
+
+func userDataStateFunc(v interface{}) string {
+	switch s := v.(type) {
+	case string:
+		return utils.Base64EncodeIfNot(s)
+	default:
+		return ""
+	}
+}
+
+func (r ContainerRegistryTaskResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ContainerRegistryTaskName,
+		},
+		"container_registry_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.RegistryID,
+		},
+		"platform_setting": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"os": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.Windows),
+							string(legacyacr.Linux),
+						}, false),
+					},
+					"architecture": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.Amd64),
+							string(legacyacr.Arm),
+							string(legacyacr.Arm64),
+							string(legacyacr.ThreeEightSix),
+							string(legacyacr.X86),
+						}, false),
+					},
+					"variant": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.V6),
+							string(legacyacr.V7),
+							string(legacyacr.V8),
+						}, false),
+					},
+				},
+			},
+		},
+		"docker_step": {
```

Comment:
step seems redundant given there can only be one? ie this is a task.docker.ect

Reply:
```suggestion
		"docker": {
```

Reply:
I prefer to keep the "step" here and the same for the other two types of steps, so to be more explicit that this block is about the configurations of the steps. Otherwise, the "docker", "file", "encoded" are not so obvious that they are related to the step, which might cause users to take some time to figure them out. WDYT?

Reply:
👍